pkg/algorithm/cstorpoolselect/v1alpha1/select_test.go

```diff
@@ -168,9 +164,9 @@ func TestNewSelection(t *testing.T) {
 
 func TestAntiAffinityFilter(t *testing.T) {
 	tests := map[string]struct {
-		CVRUid, poolUids, expectedUids []string
-		labelSelector                  string
-		expectError                    bool
+		CVRPoolUid, poolUids, expectedUids []string
```

Comment:
similar change in name might be required at other test functions (like `TestPreferredAntiAffinityFilter`) as well right?

Reply:
```go
func fakeCVRListOk(uids ...string) cvrListFn {
  return func(namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
    l := &apis.CStorVolumeReplicaList{}
    for _, uid := range uids {
        l.Items = append(l.Items,
            apis.CStorVolumeReplica{
                ObjectMeta: metav1.ObjectMeta{
                    Labels: map[string]string{
                        string(cstorPoolUIDLabelKey): uid,
                    },
                },
            },
        )
    }
    return l, nil
  }
}

func fakeCVRListErr() cvrListFn {
  return func(namespace string, opts metav1.ListOptions) (*apis.CStorVolumeReplicaList, error) {
    return nil, errors.New("fake error")
  }
}

func TestAntiAffinityFilter(t *testing.T) {
    tests := map[string]struct {
        cvrList cvrListFn
        availablePools, expectedPools []string
    }{
        "Test 1": {
            cvrList:        fakeCVRListOk("uid 1"),
            availablePools: []string{"uid 1", "uid 2", "uid 3"},
            expectedPools: []string{"uid 1", "uid 2", "uid 3"},
        },
        "Test 2": {
            cvrList:        fakeCVRListOk("uid 4", "uid 2", "uid 7"),
            availablePools: []string{"uid 6"},
            expectedPools: []string{"uid 6"},
        },
        "Test 3": {
            cvrList:        fakeCVRListErr(),
            availablePools: []string{"uid 6"},
            expectedPools:  []string{},
        },
    }
    for name, test := range tests {
        t.Run(name, func(t *testing.T) {
            a := antiAffinityLabel{
                labelSelector: "should not be empty",
                cvrList: test.cvrList,
            }
            actual, _ := a.filter(test.providedPools)
            if len(test.expectedPools) != len(actual) {
                t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedPools, actual)
            } 
            if len(output) != 0 && !reflect.DeepEqual(test.expectedPools, output) {
                t.Fatalf("test %q failed: expected %v but got %v", name, test.expectedPools, actual)
            }
        })
    }
}
```

Reply:
made the changes